proof-producer/CMakeLists.txt

```diff
@@ -20,14 +20,24 @@ include(CMConfig)
 include(CMDeploy)
 include(CMSetupVersion)
 
+option(PROOF_PRODUCER_STATIC_BINARIES "Link proof-producer binaries statically" OFF)
+
+if(PROOF_PRODUCER_STATIC_BINARIES)
+    set(BUILD_SHARED_LIBS OFF)
+    set(Boost_USE_STATIC_LIBS ON)
+else()
+    set(BUILD_SHARED_LIBS OFF)  # TODO: fix invalid-pch issue with precompiled headers to build shared libraries
```

Comment:
build dynamic assigner library works ok?

Reply:
Yes, the problem is with linking of binaries.